IDrive e2 backup documentation (#40499)
home-assistant/home-assistant.github.io: source/_integrations/idrive_e2.markdown

---
title: IDrive e2
description: Instructions on how to set up an IDrive e2 bucket to use as a backup location.
ha_release: 2025.9
ha_category:
  - Backup
ha_iot_class: Cloud Push
ha_config_flow: true
ha_domain: idrive_e2
ha_codeowners:
  - "@patrickvorgers"
ha_integration_type: service
ha_quality_scale: bronze
---

The **IDrive e2** {% term integration %} allows you to use an [IDrive e2](https://www.idrive.com/s3-storage-e2/) bucket with Home Assistant Backups.

## Prerequisites

This integration requires an existing IDrive e2 bucket and an Access Key with permissions to that bucket. For security reasons, it is strongly recommended to scope the Access Key policy as narrowly as possible to the required operations and resources only.

> {% details "Create a new IDrive e2 bucket" %}

Reviewer: documenting the IDrive side of things is a bit out of scope for Home Assistant documentation. Is there IDrive documentation we can link to instead?

Author: I used the AWS S3 documentation as a starting point and there this sentence is also added. The IDrive e2 functionality is closely related.

[AWS S3 documentation](https://www.home-assistant.io/integrations/aws_s3/)

If needed:
IDrive e2 has a [tutorial](https://www.idrive.com/s3-storage-e2/videos) section where we also could refer to.

Reviewer: I see, that makes sense. But I still think it's better to refer to their proper documentation and only mention the steps or settings that are specific to the Home Assistant setup. There is less risk that the docs here are outdated when they change their UI.

Author: Updated it in [change](https://github.com/home-assistant/home-assistant.io/pull/40499/commits/8312b8fd2e753d7dbb57fde1d2e49fa59bc8bdc2)

> 1. Log in to the [IDrive e2 Management Console](https://app.idrivee2.com/dashboard/).
> 1. Navigate to **Buckets** from the Dashboard menu.
> 1. Click **Create bucket**.
> 1. Choose a unique bucket **Name** (e.g., `home-assistant-backups-123456`).
> 1. Select your preferred IDrive e2 region from the **Enabled regions** (e.g., `Frankfurt-2`).
> 1. Adjust the settings:
>    - ✅ **Files in bucket are private** (enabled by default, recommended)
>    - ⚠️ **Versioning** (optional). This lets you recover backups after Home Assistant deletes them, but it **can increase storage costs**. Disable this to allow permanent deletion based on retention settings.
> 1. Click **Create Bucket**.
> 
> Make a note of the bucket name and region — you’ll need it later.

Reviewer: ```suggestion
1. Log in to the [IDrive e2 Management Console](https://app.idrivee2.com/dashboard/).
2. Follow the steps on [creating a bucket](https://www.idrive.com/s3-storage-e2/videos) in the iDrive documentation.
3. When prompted to define a bucket name, make sure it is unique: **Name** (for example, `home-assistant-backups-123456`).
4. When adjusting the settings:
   - **Files in bucket are private**: Enabled by default. It is recommended to leave it enabled.
   - **Versioning** (optional): Lets you recover backups after Home Assistant deletes them, but it *can increase storage costs*. Disable this to allow permanent deletion based on retention settings.
5. Make a note of the bucket name and region. You’ll need it later.
```
Tiny style tweaks:
- We [avoid e.g.](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/e/eg)
- in general, we tend to avoid icons in docs, unless they are needed to refer to an icon used in the UI, as [shown in the examples](https://developers.home-assistant.io/docs/documenting/integration-docs-examples#inline-icons)
- If using an m-dash, don't add extra space around it. I'm not sure why AI does that.
- We increment list numbers.

To create an Access Key that has access to the bucket:

1. Navigate to **Access Keys** from the Dashboard menu.
1. Click **Create Access Key**.
1. Enter a name such as `home-assistant-backup`.
1. Select the same IDrive e2 region as the bucket under **Enabled regions** (e.g., `Frankfurt-2`).
1. Adjust the settings:
   - ✅ **Access Key Expiry: disabled** (disabled by default, recommended)
   - ✅ **Assign access permission: read and write** (selected by default)
     - ✅ **Allow deletion of objects** (checked by default)
     - ⚠️ **Allow bucket deletion** (must be **unchecked**; checked by default)
1. Click **Assign buckets - Select buckets**.
1. Select the **Bucket name** of the bucket created earlier.
1. Click **Create Access Key**.
1. Save the **Access Key ID** and **Secret Access Key** and keep them safe — these will be required when configuring the integration in Home Assistant.

{% enddetails %}

{% note %}

- Avoid using credentials that have more permissions than is necessary.
- By limiting credentials to a specific bucket, you reduce risk and help keep your IDrive e2 account secure.

{% endnote %}

{% include integrations/config_flow.md %}

{% configuration_basic %}

> Access Key ID:

Reviewer: ```suggestion
Access Key ID:
```
we use [sentence-style capitalization](https://learn.microsoft.com/en-us/style-guide/capitalization) also in headings and UI strings

Author: Am I missing something because the suggestion is the same as the original text?

Author: So it should be: Access key id?

Author: ```suggestion
Access key id:
```

  description: "The access key ID for your IDrive e2 account."

> Secret Access Key:

Reviewer: ```suggestion
Secret access key:
```

  description: "The secret access key for your IDrive e2 account."

> Bucket Name:

Reviewer: ```suggestion
Bucket name:
```

  description: "IDrive e2 bucket name to store the backups. Bucket must already exist and be writable by the provided credentials."
{% endconfiguration_basic %}

## Setting up the IDrive e2 integration in Home Assistant

1. In Home Assistant, go to **Settings > Devices & Services**.
1. Click **Add Integration** and search for **IDrive e2**.
1. Enter the following details:
   - Access Key ID and Secret Access Key
1. Click **Submit**
   - The integration tests the connection, determines the endpoint, and retrieves available buckets.
1. Select the name of the bucket from the dropdown list

> 1. Click **Submit**

Reviewer: ```suggestion
```
This is covered by the config flow

Author: I used the AWS S3 documentation as a starting point and there these steps are also included. The IDrive e2 functionality is closely related. How to proceed?

[AWS S3 documentation](https://www.home-assistant.io/integrations/aws_s3/)
